gcp/gcp.go

```diff
@@ -487,6 +487,93 @@ const (
 	Owner   Role = "OWNER"
 )
 
+// Search calls google.com/cloudprint/search to query printers owned by the user/robot.
+func (gcp *GoogleCloudPrint) Search(q, atype, connection_status string) (map[string]lib.Printer, error) {
+	var client *http.Client
+	if gcp.userClient != nil {
+		client = gcp.userClient
+	} else {
+		client = gcp.robotClient
+	}
+	form := url.Values{}
+	if q != "" {
+		form.Set("q", q)
+	}
+	if atype != "" {
+		form.Set("type", atype)
+	}
+	if connection_status != "" {
+		form.Set("connection_status", connection_status)
+	}
+	form.Set("extra_fields", "semanticState")
```

Comment:
There are other extra_fields that might be useful here, just to make sure you're aware. Maybe later, if they are useful. QueuedJobsCount, for example.